api/types/role.go

```diff
@@ -334,6 +339,27 @@ func (r *RoleV3) SetAccessRequestConditions(rct RoleConditionType, cond AccessRe
 	}
 }
 
+// GetAccessReviewConditions gets conditions for access reviews.
+func (r *RoleV3) GetAccessReviewConditions(rct RoleConditionType) AccessReviewConditions {
+	cond := r.Spec.Deny.ReviewRequests
+	if rct == Allow {
+		cond = r.Spec.Allow.ReviewRequests
+	}
+	if cond == nil {
+		return AccessReviewConditions{}
+	}
+	return *cond
+}
+
+// SetAccessReviewConditions sets allow/deny conditions for access reviews.
+func (r *RoleV3) SetAccessReviewConditions(rct RoleConditionType, cond AccessReviewConditions) {
+	if rct == Allow {
+		r.Spec.Allow.ReviewRequests = &cond
+	} else {
+		r.Spec.Deny.ReviewRequests = &cond
+	}
+}
+
```

Comment:
True, but 28 of the `Role` type's 33 methods make the assumption that `RoleConditionType` is a binary value (and since it is an alias of `bool`, that is currently a safe assumption).  If we decide to change that assumption, I think it is something that should be done as a separate refactor of the `Role` API.